src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -860,8 +860,15 @@ public void updateFromTopic(Topic topic) {
                     lastModifiedTime = Coerce.toLong(topic);
                     break;
                 case PERSISTED_CURRENT_PROCESSING_FILE_HASH:
-                    //TODO: the scenario of upgrading from older version needs to be handled in next PR.
+                    // If upgrade from older version, then the fileHash does not exist but fileName exists, then
+                    // transfer from fileName to fileHash.
+                    String savedName = Coerce.toString(PERSISTED_CURRENT_PROCESSING_FILE_NAME);
+                    String savedHash = Coerce.toString(topic);
                     fileHash = Coerce.toString(topic);
+                    if (savedHash.isEmpty() && !savedName.isEmpty()) {
```

Comment:
Better to check for null given there might be no value for that topic `Objects.isNull`